src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -493,22 +515,43 @@ private int GetTotalLocalOffset()
         private int GetTotalParameterOffset()
         {
             int offset = 0;
-            for (int i = 0; i < _method.Signature.Length; i++)
+            for (int i = 0; i < _signature.Length; i++)
             {
-                offset += _method.Signature[i].GetElementSize().AsInt;
+                offset += _signature[i].GetElementSize().AsInt;
             }
+            if(!_signature.IsStatic)
+            {
+                offset += _thisType.GetElementSize().AsInt;
```

Comment:
Is this the right adjustment if `_thisType` is a value type? (Same question for other places that do `_thisType.GetElementSize()`.)

Reply:
Good catch, fixed.